src/actions.js

```diff
@@ -350,52 +366,81 @@ export const startWallet = (words, pin) => async (dispatch) => {
     }
   });
 
+  const handlePartialUpdate = async (updatedBalanceMap) => {
+    const tokens = Object.keys(updatedBalanceMap);
+    const tokensHistory = {};
+    const tokensBalance = {};
+
+    for (const token of tokens) {
+      /* eslint-disable no-await-in-loop */
+      const balance = await wallet.getBalance(token);
+      const tokenBalance = balance[0].balance;
+      tokensBalance[token] = {
+        available: tokenBalance.unlocked,
+        locked: tokenBalance.locked,
+      };
+      const history = await wallet.getTxHistory({ token_id: token });
+      tokensHistory[token] = history.map((element) => mapTokenHistory(element, token));
+      /* eslint-enable no-await-in-loop */
+    }
+
+    dispatch(partiallyUpdateHistoryAndBalance({ tokensHistory, tokensBalance }));
+  };
+
   wallet.start({ pinCode: pin, password: pin }).then((serverInfo) => {
     walletUtil.storePasswordHash(pin);
     walletUtil.storeEncryptedWords(words, pin);
+
     dispatch(setServerInfo({ version: null, network: networkName }));
 
-    if (!useWalletService) {
-      wallet.on('new-tx', (tx) => {
-        fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
-          if (updatedBalanceMap) {
-            dispatch(newTx(tx, updatedBalanceMap));
-          }
-        });
-      });
+    wallet.on('new-tx', (tx) => {
+      fetchNewTxTokenBalance(wallet, tx).then(async (updatedBalanceMap) => {
+        if (updatedBalanceMap) {
+          dispatch(newTx(tx, updatedBalanceMap));
 
-      wallet.on('update-tx', (tx) => {
-        fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
-          if (updatedBalanceMap) {
-            dispatch(updateTx(tx, updatedBalanceMap));
+          if (useWalletService) {
+            handlePartialUpdate(updatedBalanceMap);
           }
-        });
+        }
       });
+    });
 
-      wallet.conn.on('best-block-update', (height) => {
-        fetchNewHTRBalance(wallet).then((data) => {
-          if (data) {
-            dispatch(updateHeight(height, data));
+    wallet.on('update-tx', (tx) => {
+      fetchNewTxTokenBalance(wallet, tx).then((updatedBalanceMap) => {
+        if (updatedBalanceMap) {
+          dispatch(updateTx(tx, updatedBalanceMap));
+
+          // Read comment on new-tx event
```

Comment:
What comment?

Reply:
This code has been refactored, this comment no longer exists